### PR TITLE
Testst by dao

### PR DESCRIPTION
**Pull request by Raychenko88**

### CODE REVIEW

File: src/main/java/info/sjd/controller/CartController.java [hunk 1]
```diff
@@ -65,10 +65,9 @@
 
     @PostMapping(path = "update_status")
     public ResponseEntity<Cart> updateStatus(@RequestParam Integer idParam,@RequestParam Integer closedParam){
-        Cart cart = cartService.updateStatus(idParam, closedParam);
-        if (cart != null){
-            return new ResponseEntity<>(cartService.updateStatus(idParam,closedParam), HttpStatus.OK);
+        if (updateStatus(idParam, closedParam) != null){
```
Comment (Ignatenko2207) [inline src/main/java/info/sjd/controller/CartController.java]: What is it?! You call the same method in this method recursively? Why? 
You should get results from service and check it here to send correct response.

File: src/main/java/info/sjd/controller/CartController.java [hunk 2]
```diff
@@ -65,10 +65,9 @@
...
+            cartService.updateStatus(idParam,closedParam);
         }
-        return new ResponseEntity<>(HttpStatus.NOT_ACCEPTABLE);
+        return new ResponseEntity<>(HttpStatus.OK);
```
Comment (Ignatenko2207) [inline src/main/java/info/sjd/controller/CartController.java]: You always send OK status. But you need check results before.

File: src/main/java/info/sjd/controller/OrderController.java [hunk 3]
```diff
@@ -59,12 +59,11 @@
     }
 
     @GetMapping(path = "update_amount")
-    public ResponseEntity<Order> updateAmount(@RequestParam Integer id, Integer amount){
-        Order order = orderService.updateAmount(id, amount);
-        if (order != null){
-            return new ResponseEntity<>(orderService.updateAmount(id,amount), HttpStatus.OK);
+    public ResponseEntity<Order> updateAmount(@RequestParam Integer id,@RequestParam Integer amount){
```
Comment (Ignatenko2207) [inline src/main/java/info/sjd/controller/OrderController.java:62]: The same situation as before in updateStatus method

File: src/main/java/info/sjd/controller/OrderController.java [hunk 4]
```diff
@@ -59,12 +59,11 @@
...
+        if (updateAmount(id, amount) != null){
+            orderService.updateAmount(id,amount);
         }
-        return new ResponseEntity<>(HttpStatus.NOT_ACCEPTABLE);
+        return new ResponseEntity<>(HttpStatus.OK);
```
Comment (Ignatenko2207) [inline src/main/java/info/sjd/controller/OrderController.java]: Response is always OK. It's not correct

File: src/test/java/info/sjd/dao/CartDAOTest.java [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+        carts.add(cart);
+        assertNotNull(saveCart);
+        assertEquals(0, saveCart.getClosed());
+
+        cartDAO.updateStatus(cart.getId(), 1);
+        Cart cartUpdate = cartDAO.findById(cart.getId()).orElse(null);
+        assertNotNull(cartUpdate);
+        assertEquals(1, cartUpdate.getClosed());
+    }
+}
```
Comment (Ignatenko2207) [inline src/test/java/info/sjd/dao/CartDAOTest.java]: Add new line

File: src/test/java/info/sjd/dao/ItemDAOTest.java [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+        Item item2 = Item.builder().name("t_name").code("t_code").price(321).availability(2).build();
+        itemDAO.save(item1);
+        itemDAO.save(item2);
+        List<Item> list = itemDAO.getAllAvailable();
+        assertNotNull(list);
+        assertFalse(list.isEmpty());
+        itemDAO.delete(item1);
+        itemDAO.delete(item2);
+    }
+}
```
Comment (Ignatenko2207) [inline src/test/java/info/sjd/dao/ItemDAOTest.java]: Add new line

File: src/test/java/info/sjd/dao/OrderDAOTest.java [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+        Order order =  Order.builder().item(item).cart(cart).amount(50).build();
+        orderDAO.save(order);
+        orders.add(order);
+        assertNotNull(order.getId());
+
+        Order targetOrder = orderDAO.findOrderByItem(item.getId());
+        assertNotNull(targetOrder);
+        assertEquals(item.getId(), targetOrder.getItem().getId());
+    }
+}
```
Comment (Ignatenko2207) [inline src/test/java/info/sjd/dao/OrderDAOTest.java]: Add new line

File: src/test/java/info/sjd/dao/UserDAOTest.java [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+    @Test
+    void findByLoginAndPasswordTest() {
+        User user1 = User.builder().login("test_login1").password("test_pass1").firstName("test_fn").lastName("test_ln").build();
+        userDAO.save(user1);
+        User user = userDAO.findByLoginAndPassword(user1.getLogin(), user1.getPassword());
+        assertNotNull(user);
+        assertEquals(user1.getLogin(),user.getLogin());
+        userDAO.delete(user1);
+    }
+}
```
Comment (Ignatenko2207) [inline src/test/java/info/sjd/dao/UserDAOTest.java]: Add new line

File: src/test/resources/application.yml [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+      driver-class-name: "org.h2.Driver"
+  jpa:
+    show-sql: true
+    hibernate:
+      ddl-auto: update
+    properties:
+      hibernate:
+        dialect: "org.hibernate.dialect.H2Dialect"
+#    profiles:
+#      active: rest
```
Comment (Ignatenko2207) [inline src/test/resources/application.yml]: Add new line

File: src/main/java/info/sjd/controller/CartController.java (hunk 1 above)
Comment (Raychenko88) [reply src/main/java/info/sjd/controller/CartController.java]: CarConrollert updateStatus done, and OrderConrollert updateAmount done

File: src/main/java/info/sjd/controller/CartController.java (hunk 2 above)
Comment (Raychenko88) [reply src/main/java/info/sjd/controller/CartController.java]: CarConrollert updateStatus done, and OrderConrollert updateAmount done

File: src/main/java/info/sjd/controller/OrderController.java (hunk 3 above)
Comment (Raychenko88) [reply src/main/java/info/sjd/controller/OrderController.java:62]: done

File: src/main/java/info/sjd/controller/OrderController.java (hunk 4 above)
Comment (Raychenko88) [reply src/main/java/info/sjd/controller/OrderController.java]: done

File: src/test/java/info/sjd/dao/CartDAOTest.java (hunk 5 above)
Comment (Raychenko88) [reply src/test/java/info/sjd/dao/CartDAOTest.java]: done

File: src/test/java/info/sjd/dao/ItemDAOTest.java (hunk 6 above)
Comment (Raychenko88) [reply src/test/java/info/sjd/dao/ItemDAOTest.java]: done

File: src/test/java/info/sjd/dao/OrderDAOTest.java (hunk 7 above)
Comment (Raychenko88) [reply src/test/java/info/sjd/dao/OrderDAOTest.java]: done

File: src/test/resources/application.yml (hunk 9 above)
Comment (Raychenko88) [reply src/test/resources/application.yml]: done

**Review by Ignatenko2207 — APPROVED** [review]

File: src/test/java/info/sjd/dao/UserDAOTest.java (hunk 8 above)
Comment (Raychenko88) [reply src/test/java/info/sjd/dao/UserDAOTest.java]: done

**Pull request merged**